### ISSUE TITLE
Fix CLI doc exit code docs

### ISSUE THREAD

**Issue by ybiquitous** [stylelint/stylelint#4546]
The CLI document says about the exit codes as follow:

https://github.com/stylelint/stylelint/blob/90bef807ac270ae4082384de71ada7f56bbb3145/docs/user-guide/cli.md#L247-L254

However, the current stylelint never exits with the code `80`. The following commands can reproduce the issue:

```shell-session
$ ./node_modules/.bin/stylelint --version
13.0.0

$ ./node_modules/.bin/stylelint '*.css'
Error: No files matching the pattern "*.css" were found.
    at /private/tmp/work/node_modules/stylelint/lib/standalone.js:208:12
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

$ echo $?
1
```

This issue does NOT occur on stylelint v7:

```shell-session
$ ./node_modules/.bin/stylelint --version
7.13.0

$ ./node_modules/.bin/stylelint '*.css'
Error: *.css does not match any files
    at /private/tmp/work/node_modules/stylelint/lib/standalone.js:124:33

$ echo $?
80
```

And, stylelint v8 exits with the code `0`:

```shell-session
$ ./node_modules/.bin/stylelint --version
8.4.0

$ ./node_modules/.bin/stylelint '*.css'

$ echo $?
0
```

About this issue, which is correct the document or the current behavior?

Other, I noticed that Running stylelint with invalid options exits with the code `2` instead of `1`:

```shell-session
$ ./node_modules/.bin/stylelint --foo
Invalid option "--foo". Did you mean "-f"?

$ echo $?
2
```

I think appropriate to exit with the code `1` because passing invalid options is not a violation of rules.

I think we should reconsider the document and code about exit codes and add more system test cases. What do you think about it?

**Comment by jeddy3** [comment]
I went down the history rabbit hole with this one!

The original exit codes were added in https://github.com/stylelint/stylelint/pull/380#issuecomment-139700875, which refers to https://github.com/sds/scss-lint#exit-status-codes and in turn refers to http://www.gsp.com/cgi-bin/man.cgi?section=3&topic=sysexits. There's also https://github.com/stylelint/stylelint/issues/26#issuecomment-132153608 with more exit codes.

We started with exit code 80 when no files were found. Then changed the behaviour to not error. Then reinstated the behaviour in `10.0.0`, but with exit code 1.

I think the current behaviour is good enough and the docs need updating. I struggled to find any reference to exit code 80 outside of those SCSSLint docs. The current behaviour is also consistent with ESLint:

```console
jeddy3@jeddy3 % npx eslint test.css
No files matching 'test.css' were found.
jeddy3@jeddy3 % echo $?            
1
jeddy3@jeddy3 % npx eslint --rule s 
Invalid value for option 'rule' - expected type Object, received value: s.
jeddy3@jeddy3 % echo $?             
2
```

We could go with:

> -   1: Something unknown went wrong. 
> -   2: At least one rule or CLI flag error 
> -   78: There was some problem with the configuration file. 

A bonus of no breaking changes.
